lib/galaxy/files/sources/posix.py

```diff
@@ -39,7 +39,8 @@ def _list(self, path="/", recursive=True, user_context=None):
         if recursive:
             res = []
             for (p, dirs, files) in safe_walk(dir_path, allowlist=self._allowlist):
-                rel_dir = os.path.relpath(p, dir_path)
+                dir_path_root = os.path.split(dir_path)[0]
```

Comment:
> Maybe consider moving this out of the loop? It needs to run only once.

Hey @ic4f. I'm not sure :sweat_smile:
how would it work without a loop? We need to walk among sub-directories as well, right? that's what loop does (at least in my understanding). Is there a way to avoid the loop?

Reply:
Oh, now I see the `p` argument! You're right - sorry! I need more coffee...

Reply:
Wait, I'm not crazy after all. I was referring to the code I saw in the **previous** commit - 
```
dir_path_root = os.path.split(dir_path)[0]
```
That's the line that got executed on each iteration, but did the same thing. But now I see the new version doesn't include that line - so it's irrelevant now.

Reply:
> Oh, now I see the `p` argument! You're right - sorry! I need more coffee...

:beers:

Reply:
> Wait, I'm not crazy after all. I was referring to the code I saw in the **previous** commit -
> 
> ```
> dir_path_root = os.path.split(dir_path)[0]
> ```
> 
> That's the line that got executed on each iteration, but did the same thing. But now I see the new version doesn't include that line - so it's irrelevant now.

aha. Ok then, that was the right thing to do :+1: